### DISCUSSION TITLE
attribute types

### DISCUSSION THREAD

**Discussion by seethroughdev** [ts-graphviz/ts-graphviz#802] [Q&A] [answered]
Really liking this library so far!  Thanks for putting it together.

Question on supporting attributes.  It doesn't seem the types support all options in: https://www.graphviz.org/doc/info/attrs.html.

Not sure if that's intentional or not.  Is this something I can add a PR to add more options and documentation?

Also not sure what the preferred method for setting attributes on an object is.  For now I'm doing: `g.attributes.node.set("shape", "square")`.  But of course it would be nice to set an Options object for multiple values at once.

Thanks again!

**Comment by kamiazya** [comment, marked as answer]
Hi,

Thank you for using my library and noticing the flaws!

PR is welcome!

To add types and documentation for attributes, edit the file.

https://github.com/ts-graphviz/ts-graphviz/blob/master/src/attribute.ts


> Also not sure what the preferred method for setting attributes on an object is. For now I'm doing: g.attributes.node.set("shape", "square"). But of course it would be nice to set an Options object for multiple values at once.

Also, as a method to specify attributes at once, an `apply` method is prepared.

https://ts-graphviz.github.io/ts-graphviz/classes/attributes.html#apply

You call like this.

```ts
g.attributes.node.apply({
  shape: 'square',
});
```

Thanks!

**Comment by kamiazya** [comment]
@seethroughtrees 

The issue will be closed as if the question was resolved.

PR is always welcome!

**Comment by kamiazya** [comment]
@all-contributors please add @seethroughdev  for question.

[a comment by an automated account is not shown]
